### ISSUE TITLE
Задача 2.2.23. Experiments: Improvements.

### ISSUE THREAD

**Issue by be9** [v-log/Training#50]
Run empirical studies to evaluate the effectiveness of each of the three improvements to mergesort that are described in the text (see Exercise 2.2.11, #47). ~~Also, compare the performance of the merge implementation given in the text with the merge described in Exercise 2.2.10.~~ In particular, empirically determine the best value of the parameter that decides when to switch to insertion sort for small subarrays.

(зачёркнутое делать не надо).

**Comment by v26** [comment]
@be9 Здесь у меня вопрос такой:
Допустим, я хочу сделать универсальный метод timer, которому бы как-то передавался любой метод с аргументами этого метода, чтобы засечь время выполнения. Можно ли это реализовать? (Если можно, не говорите как, просто можно ли)

**Comment by be9** [comment]
Конечно, можно. Для этого используется конструкция, с которой ты уже хорошо знаком, позволяющая передать в метод как бы кусок кода, который тот будет вызывать.
